client/scala/scala-armada-client/src/test/scala/ArmadaClientSuite.scala

```diff
@@ -103,8 +135,12 @@ private class JobsMockServer extends JobsGrpc.Jobs {
   }
 
   def getJobStatus(request: JobStatusRequest): scala.concurrent.Future[JobStatusResponse] = {
-    val response = new JobStatusResponse(Map("fakeJobId" -> JobState.RUNNING))
-    Future.successful(response)
+    val statusMap = collection.mutable.Map[String,JobState]() // jobID -> state
+    val it = jobMap.keys.asIterator()
+
+    while (it.hasNext()) statusMap.put(it.next(), JobState.RUNNING)
+
```

Comment:
I'm still learning, but I think something like this is more scala like:
```
    jobMap.keySet.forEach(k => statusMap.put(k, JobState.RUNNING))
```

Reply:
Good suggestion - I wasn't clear if the iterator pattern or the enumerator pattern was the now-preferred version.  Fixed in  commit 5d569f7d .